wgsl/index.bs

```diff
@@ -338,6 +338,25 @@ Issue: (dneto): Complete description of `Array<E,N>`
   </xmp>
 </div>
 
+<table class='data'>
+  <caption>Structure attributes</caption>
+  <thead>
+    <tr><th>Attribute<th>Description
+  </thead>
+
+  <tr><td><dfn noexport>`block`</dfn>
+      <td>Applies to a structure type.<br>
+         Indicates this structure type represents the contents of a
+         buffer resource occupying a single binding slot in the shader's resource
+         interface.
+         The `block` attribute must be applied to a structure type used
+         as the [=store type=] of a [=uniform buffer=] or [=storage buffer=] variable.
+</table>
+
+A structure type with the [=block=] attribute must not be the element type of an array type.
+
+A structure type with the [=block=] attribute must not be the member type in a structure type.
```

Comment:
I was going for using "must not" so that it's easier to find as a validation rule.

I'm planning to reword this as:


> A structure type with the [=block=] attribute must not be:
> * the element type of an array type.
> *  the member type in another structure.